services/worker-manager/src/providers/aws.js

```diff
@@ -254,6 +252,37 @@ class AwsProvider extends Provider {
     }));
   }
 
+  async removeWorker({worker}) {
+    let result;
+    try {
+      result = await this.ec2s[worker.providerData.region].terminateInstances({
+        InstanceIds: [worker.workerId],
+      }).promise();
+    } catch (e) {
+      const workerPool = this.WorkerPool.load({
+        workerPoolId: worker.workerPoolId,
+      });
+      await workerPool.reportError({
```

Comment:
Ah, my comment here became outdated. I think we should check to see if the instance is already gone here and if so, not report an error. The requested state is already true in that case

Reply:
If instance is already gone, the aws request won't error, it will be considered a success